instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/KubernetesExperimentalAttributesExtractor.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.kubernetesclient;
+
+import static io.opentelemetry.javaagent.instrumentation.kubernetesclient.KubernetesClientSingletons.captureExperimentalSpanAttributes;
+
+import io.kubernetes.client.openapi.ApiResponse;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import okhttp3.Request;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+class KubernetesExperimentalAttributesExtractor
+    extends AttributesExtractor<Request, ApiResponse<?>> {
+  @Override
+  protected void onStart(AttributesBuilder attributes, Request request) {
+    if (!captureExperimentalSpanAttributes()) {
```

Comment:
If experimental attributes are turned off you could just not add this `AttributesExtractor` to the instrumenter.